text/0000-service-hook.md

```diff
@@ -0,0 +1,100 @@
+* Start Date: 2019-03-21
+* RFC PR: (leave this empty)
+* Fusion Issue: (leave this empty)
+
+# Summary
+
+Add hook for consuming a registered service
+
+
+# Basic example
+
+We have a plugin registered to a token. We want to use the service that plugin provides.
+
+```javascript
+// components/Example.js
+import {ExampleToken} from 'fusion-tokens';
+import {useService} from 'fusion-react';
+
+export default function Example() {
+  const service = useService(ExampleToken);
+
+  return (
+    <button onClick={service}>Invoke service</button>
+  );
+}
+```
+
+# Motivation
+
+Consuming services through the legacy Context API is cumbersome and increasingly difficult with the adoption of hooks in Fusion. This proposal aims to provide a simplified and contemporary way of providing services to a React application.
+
+This work will not include refactoring existing plugins to hooks but will allow the migration path for all current '-react' plugins in the near future.
+
+# Detailed design
+
+The `useService` hook would get a function from Context to lookup the registered service for the supplied token.
+
+```javascript
+// fusion-react
+const ServiceContext = React.createContext();
+
+export function useService(token) {
```

Comment:
this looks good, although it might be useful to include the flow types here.

Reply:
You caught me, I was avoiding that. Will update